File: c#/str_join_slow.cs
using System;
using System.Linq;
using System.Diagnostics;

class Program {
    const string Chunk = "regexp123abcbcd";
    const int RepeatCount = 10000;
    const int Size = 1000;

    static void Join() {
        string.Concat(Enumerable.Repeat(Chunk, Size));
    }

    static void Main() {
        var sw = new Stopwatch();
        sw.Start();
        for (var i = 0; i < RepeatCount; ++i) Join();
        sw.Stop();
        Console.WriteLine(sw.ElapsedMilliseconds);
    }
}

File: lua/str_join_slow.lua
local chunk = "regexp123abcbcd"
local repeatCount = 10000
local size = 1000

function join()
  str = ""
  for i = 1, size do
    str = str .. chunk
  end
end

local start = os.clock()
for i = 1, repeatCount do
  join()
end
local _end = os.clock()

print((_end - start) * 1000)
